### DISCUSSION TITLE
First Timer In Need of Some Help

### DISCUSSION THREAD

**Discussion by HarrisonCreates** [bigcapitalhq/bigcapital#205] [Q&A]
Hello everyone, apologies if it's obvious, but this is my first time using Docker to deploy an application and the docs seem incomplete for a beginner like me to get this up and running. In the Architecture part of the docs, it mentions MongoDB and MySQL, but in the System Recommendations section, it mentions MariaDB. Do I need to have three databases installed at once for this to work? This looks like a truly amazing project and I'm excited to get it up and running, I'm just hitting a few speedbumps along the way. Also, is there a boilerplate nginx server block that can be used for the requirement that reads: "Proxy server configured to redirect requests that start with /api to dynamic data (API endpoints) to the server, and all other requests to static assets of the single-page application." One final question: what is Agendash, and what is the utility of its configuration lines in the .env file? Apologies again if it's obvious, but I searched the docs and couldn't find answers to these questions. If I missed something, please link me and I'd be happy to take a look 😄

**Comment by xprnio** [comment]
> In the Architecture part of the docs, it mentions MongoDB and MySQL, but in the System Recommendations section, it mentions MariaDB. Do I need to have three databases installed at once for this to work?

MariaDB is a fork of MySQL, and if you have a look at the [docker-compose.yml](https://github.com/bigcapitalhq/bigcapital/blob/develop/docker-compose.yml#L9) you'll see it configured there. I _may_ be wrong on this, but I think that you _might be_ able to use either one, but MariaDB is definitely the one you should go for unless there's a reason why you can't.

> Also, is there a boilerplate nginx server block that can be used for the requirement that reads: "Proxy server configured to redirect requests that start with /api to dynamic data (API endpoints) to the server, and all other requests to static assets of the single-page application."

Yes, you can find the server block [here](https://github.com/bigcapitalhq/bigcapital/blob/develop/docker/nginx/sites/server.template).

> What is Agendash, and what is the utility of its configuration lines in the .env file?

Agendash is a dashboard for Agenda, the system used for scheduling (background) jobs. You can use the user and password defined there to access `http://localhost:3000/agendash/dash` (or `/api/agendash/dash` if you're running BigCapital behind the reverse proxy).
